src/sentry/models/organization.py

```diff
@@ -188,12 +191,20 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         if not self.id:
-            self.id = snowflake_id_generation()
+            self.id = snowflake_id_generation(self.snowflake_redis_key)
         if not self.slug:
             lock = locks.get("slug:organization", duration=5)
             with TimedRetryPolicy(10)(lock.acquire):
                 slugify_instance(self, self.name, reserved=RESERVED_ORGANIZATION_SLUGS)
-        super().save(*args, **kwargs)
+        try:
+            with transaction.atomic():
+                super().save(*args, **kwargs)
+        except IntegrityError:
+            if self.snowflake_retry_counter == settings.MAX_REDIS_SNOWFLAKE_RETY_COUNTER:
+                raise Exception("Max allowed ID retry reached. Please try again in a second")
+            self.snowflake_retry_counter += 1
```

Comment:
This never gets reset to 0? Intuitively I'd expect it to retry that many times per (non-recursive) call to the `save` method, not for the lifecycle of the model instance.

If the "Max allowed ID retry reached" exception gets raised, the object will just be unsaveable forever. It seems like we're assuming that the exception will cause the object to be discarded and whatever operation queried for it will start over fresh, querying for a new instance of the model class. But if the retry operation keeps the same model instance in memory, the caller is in for an unpleasant surprise.

I'd rather keep the retry counter as a local variable that's initialized to 0 every time `save` is called from the outside. That would require refactoring the recursive call into a while loop.